pkg/clustermanager/manager.go

```diff
@@ -315,6 +317,16 @@ func ToRESTConfig(cluster *v3.Cluster, context *config.ScaledContext) (*rest.Con
 					ht.DialTLS = tlsDialer
 				}
 			}
+			if cluster.Status.Driver == "googleKubernetesEngine" {
+				ts, err := gke.GetTokenSource(context.RunContext, (*cluster.Spec.GenericEngineConfig)["credential"].(string))
```

Comment:
`spec.GenericEngineConfig` is a pointer, we should check for nil otherwise you can get a panic here. Possible that driver is set but spec is not.